airflow/providers/airbyte/operators/helper.py

```diff
@@ -0,0 +1,98 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional
+
+
+def map_airbyte_type(type: str, format: Optional[str]) -> str:
+    if type == "string" and format == "date-time":
+        return "timestamp"
+
+    if type == "string" and format == "date":
+        return "date"
+
+    if type == "string":
+        return "string"
+    if type == "integer":
+        return "int"
+    if type == "number":
+        return "float"
+    if type == "boolean":
+        return "boolean"
+    if type == "object":
+        return "object"
+    if type == "array":
+        return "array"
+    return "string"
+
+
+def get_field_type(field_metadata: Dict[str, Any]) -> str:
+    airbyte_field_type = field_metadata.get("type", None)
+
+    airbyte_target_field_type = ""
+
+    if isinstance(airbyte_field_type, list):
+        airbyte_target_field_type = map_airbyte_types(airbyte_field_type, field_metadata.get("format", None))
+
+    if isinstance(airbyte_field_type, str):
```

Comment:
Shouldn't we go with one OR the other? Now both of these checks will be always triggered.
```suggestion
    elif isinstance(airbyte_field_type, str):
```

Reply:
I think I like more keeping happy path on the left, so I'll instead of elif add return statment.